src/algebra/monoid_algebra/basic.lean

```diff
@@ -243,12 +243,17 @@ instance [ring k] [semigroup G] : non_unital_ring (monoid_algebra k G) :=
 { .. monoid_algebra.add_comm_group,
   .. monoid_algebra.non_unital_semiring }
 
+-- note that whenever possible, we have to extend `non_assoc_ring` to get the `int_cast`.
+
```

Comment:
I'd lean towards dropping this comment (and the other copy of it); in general you always have to extend the direct subclass to avoid risking getting default fields by accident.